depends/patches/qt/macos_skip_version_checks.patch

```diff
@@ -0,0 +1,66 @@
+The @available compiler directive exports the __isPlatformVersionAtLeast
```

Comment:
> Where does this patch come from? If upstream, can you link to the issue/commits? If not upstream/ it's a bug in Qt, has an issue been opened upstream?

Not upstream. Qt does not officially support cross-compiling from non-macOS build platforms to macOS. I don't think any related issue upstream will be considered.

Reply:
Ok. This is interesting, because `(@available(macOS` is used in more places in the Qt codebase (and I assume this will only increase in future) than what is patched out here, and, was also present in Qt5, but we didn't have to patch it out there? So maybe there is some other build-related issue here?

This approach of essentially hardcoding our oldest supported version, as also the newest supported version, also means any users of the newer OS's, may be missing out on new features/other improvements, but I guess that is the tradeoff.

Reply:
> Ok. This is interesting, because `(@available(macOS` is used in more places in the Qt codebase (and I assume this will only increase in future) than what is patched out here, and, was also present in Qt5, but we didn't have to patch it out there?

Qt 5 codebase uses `(@available(macOS` in code that we do not use, specifically in the RHI implementation for the Metal backend (`src/gui/rhi/qrhimetal.mm`) and in iOS plugins (`plugins/platforms/ios/...`).

> So maybe there is some other build-related issue here?

I don’t think this patch could cause any of those issues.